### ISSUE TITLE
Build error on GCC 14 / Fedora Rawhide with liblo-0.32

### ISSUE THREAD

**Issue by ycollet** [amsynth/amsynth#231]
I am trying to build amsynth for Fedora rawhide.
liblo version is 0.32
gcc version is 14.0.1
The error message:
```
g++ -std=c++11 -DHAVE_CONFIG_H -I.  -DDATADIR='"/usr/share"' -DPKGDATADIR='"/usr/share/amsynth"' -DPACKAGE_LOCALEDIR=\""/usr/share/locale"\"  -fPIC -fvisibility=hidden -I"./vendor" -DGDK_DISABLE_DEPRECATED -DGSEAL_ENABLE -DGTK_DISABLE_DEPRECATED -DGTK_DISABLE_SINGLE_INCLUDES  -I/usr/include/gtk-2.0 -I/usr/lib64/gtk-2.0/include -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/webp -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -I/usr/include/libxml2 -I/usr/include/pixman-1 -I/usr/include/libpng16 -DWITH_GZFILEOP -I/usr/include/sysprof-6 -pthread   -fno-strict-aliasing -Wall -Wextra -Wundef -Wwrite-strings -Wpointer-arith -Wmissing-declarations -Wredundant-decls -Wno-unused-parameter -Wno-missing-field-initializers -Wformat=2 -Wcast-align -Wformat-nonliteral -Wformat-security -Wsign-compare -Wstrict-aliasing -Wshadow -Winline -Wpacked -Wmissing-format-attribute -Wmissing-noreturn -Winit-self -Wmissing-include-dirs -Wunused-but-set-variable -Warray-bounds -Wreturn-type -Wno-overloaded-virtual -Wswitch-enum -Wswitch-default -Wno-error=unused-parameter -Wno-error=missing-field-initializers -Wno-error=overloaded-virtual -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=x86-64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -mtls-dialect=gnu2 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer  -c -o src/amsynth_dssi_gtk-amsynth_dssi_gtk.o `test -f 'src/amsynth_dssi_gtk.cpp' || echo './'`src/amsynth_dssi_gtk.cpp
src/amsynth_dssi_gtk.cpp: In function â€˜int main(int, char**)â€™:
src/amsynth_dssi_gtk.cpp:238:86: error: invalid conversion from â€˜int (*)(const char*, const char*, lo_arg**, int, void*, void*)â€™ to â€˜lo_method_handlerâ€™ {aka â€˜int (*)(const char*, const char*, lo_arg**, int, lo_message_*, void*)â€™} [-fpermissive]
  238 |     lo_server_add_method(_osc_server, tmpstr("/%s/control",     _osc_path), "if",    osc_control_handler,     nullptr);
      |                                                                                      ^~~~~~~~~~~~~~~~~~~
      |                                                                                      |
      |                                                                                      int (*)(const char*, const char*, lo_arg**, int, void*, void*)
In file included from /usr/include/lo/lo.h:32,
                 from src/amsynth_dssi_gtk.cpp:31:
/usr/include/lo/lo_lowlevel.h:876:72: note:   initializing argument 4 of â€˜lo_method_* lo_server_add_method(lo_server, const char*, const char*, lo_method_handler, const void*)â€™
  876 |                                const char *typespec, lo_method_handler h,
      |                                                      ~~~~~~~~~~~~~~~~~~^
src/amsynth_dssi_gtk.cpp:239:86: error: invalid conversion from â€˜int (*)(const char*, const char*, lo_arg**, int, void*, void*)â€™ to â€˜lo_method_handlerâ€™ {aka â€˜int (*)(const char*, const char*, lo_arg**, int, lo_message_*, void*)â€™} [-fpermissive]
  239 |     lo_server_add_method(_osc_server, tmpstr("/%s/sample-rate", _osc_path), "i",     osc_samplerate_handler,  nullptr);
      |                                                                                      ^~~~~~~~~~~~~~~~~~~~~~
      |                                                                                      |
      |                                                                                      int (*)(const char*, const char*, lo_arg**, int, void*, void*)
/usr/include/lo/lo_lowlevel.h:876:72: note:   initializing argument 4 of â€˜lo_method_* lo_server_add_method(lo_server, const char*, const char*, lo_method_handler, const void*)â€™
  876 |                                const char *typespec, lo_method_handler h,
      |                                                      ~~~~~~~~~~~~~~~~~~^
src/amsynth_dssi_gtk.cpp:240:86: error: invalid conversion from â€˜int (*)(const char*, const char*, lo_arg**, int, void*, void*)â€™ to â€˜lo_method_handlerâ€™ {aka â€˜int (*)(const char*, const char*, lo_arg**, int, lo_message_*, void*)â€™} [-fpermissive]
  240 |     lo_server_add_method(_osc_server, tmpstr("/%s/program",     _osc_path), "ii",    osc_program_handler,     nullptr);
      |                                                                                      ^~~~~~~~~~~~~~~~~~~
      |                                                                                      |
      |                                                                                      int (*)(const char*, const char*, lo_arg**, int, void*, void*)
/usr/include/lo/lo_lowlevel.h:876:72: note:   initializing argument 4 of â€˜lo_method_* lo_server_add_method(lo_server, const char*, const char*, lo_method_handler, const void*)â€™
  876 |                                const char *typespec, lo_method_handler h,
      |                                                      ~~~~~~~~~~~~~~~~~~^
src/amsynth_dssi_gtk.cpp:241:86: error: invalid conversion from â€˜int (*)(const char*, const char*, lo_arg**, int, void*, void*)â€™ to â€˜lo_method_handlerâ€™ {aka â€˜int (*)(const char*, const char*, lo_arg**, int, lo_message_*, void*)â€™} [-fpermissive]
  241 |     lo_server_add_method(_osc_server, tmpstr("/%s/show",        _osc_path), nullptr, osc_show_handler,        nullptr);
      |                                                                                      ^~~~~~~~~~~~~~~~
      |                                                                                      |
      |                                                                                      int (*)(const char*, const char*, lo_arg**, int, void*, void*)
/usr/include/lo/lo_lowlevel.h:876:72: note:   initializing argument 4 of â€˜lo_method_* lo_server_add_method(lo_server, const char*, const char*, lo_method_handler, const void*)â€™
  876 |                                const char *typespec, lo_method_handler h,
      |                                                      ~~~~~~~~~~~~~~~~~~^
src/amsynth_dssi_gtk.cpp:242:86: error: invalid conversion from â€˜int (*)(const char*, const char*, lo_arg**, int, void*, void*)â€™ to â€˜lo_method_handlerâ€™ {aka â€˜int (*)(const char*, const char*, lo_arg**, int, lo_message_*, void*)â€™} [-fpermissive]
  242 |     lo_server_add_method(_osc_server, tmpstr("/%s/hide",        _osc_path), nullptr, osc_hide_handler,        nullptr);
      |                                                                                      ^~~~~~~~~~~~~~~~
      |                                                                                      |
      |                                                                                      int (*)(const char*, const char*, lo_arg**, int, void*, void*)
/usr/include/lo/lo_lowlevel.h:876:72: note:   initializing argument 4 of â€˜lo_method_* lo_server_add_method(lo_server, const char*, const char*, lo_method_handler, const void*)â€™
  876 |                                const char *typespec, lo_method_handler h,
      |                                                      ~~~~~~~~~~~~~~~~~~^
src/amsynth_dssi_gtk.cpp:243:86: error: invalid conversion from â€˜int (*)(const char*, const char*, lo_arg**, int, void*, void*)â€™ to â€˜lo_method_handlerâ€™ {aka â€˜int (*)(const char*, const char*, lo_arg**, int, lo_message_*, void*)â€™} [-fpermissive]
  243 |     lo_server_add_method(_osc_server, tmpstr("/%s/quit",        _osc_path), nullptr, osc_quit_handler,        nullptr);
      |                                                                                      ^~~~~~~~~~~~~~~~
      |                                                                                      |
      |                                                                                      int (*)(const char*, const char*, lo_arg**, int, void*, void*)
/usr/include/lo/lo_lowlevel.h:876:72: note:   initializing argument 4 of â€˜lo_method_* lo_server_add_method(lo_server, const char*, const char*, lo_method_handler, const void*)â€™
  876 |                                const char *typespec, lo_method_handler h,
      |                                                      ~~~~~~~~~~~~~~~~~~^
src/amsynth_dssi_gtk.cpp:244:57: error: invalid conversion from â€˜int (*)(const char*, const char*, lo_arg**, int, void*, void*)â€™ to â€˜lo_method_handlerâ€™ {aka â€˜int (*)(const char*, const char*, lo_arg**, int, lo_message_*, void*)â€™} [-fpermissive]
  244 |     lo_server_add_method(_osc_server, nullptr, nullptr, osc_fallback_handler, nullptr);
      |                                                         ^~~~~~~~~~~~~~~~~~~~
      |                                                         |
      |                                                         int (*)(const char*, const char*, lo_arg**, int, void*, void*)
/usr/include/lo/lo_lowlevel.h:876:72: note:   initializing argument 4 of â€˜lo_method_* lo_server_add_method(lo_server, const char*, const char*, lo_method_handler, const void*)â€™
  876 |                                const char *typespec, lo_method_handler h,
      |                                                      ~~~~~~~~~~~~~~~~~~^
make[2]: *** [Makefile:3366: src/amsynth_dssi_gtk-amsynth_dssi_gtk.o] Error 1
make[2]: Leaving directory '/builddir/build/BUILD/amsynth-release-1.13.2'
```

**Comment by nickdowell** [comment]
Thanks for reporting this issue, I will look into a fix.

In the meantime, you can skip building the DSSI plugin by running `./configure --with-dssi` and you should be able to still build the other targets (standalone app, LV2 & VST)

**Comment by nickdowell** [comment]
Hi @ycollet I have prepared a release branch containing a fix for this, but I don't have Fedora Rawhide to test on. If you could try out the `1.13.3` branch and let me know whether that fixes the build for you that would be very helpful, thanks 🙏

**Comment by ycollet** [comment]
Branch 1.13.3 tested on Fedora 38 and Fedora Rawhide. Worked in both case.
Thanks a lot for the fix !

**Comment by nickdowell** [comment]
Fixed in [release 1.13.3](https://github.com/amsynth/amsynth/releases/tag/release-1.13.3) :tada:
